src/backend/distributed/commands/utility_hook.c

```diff
@@ -627,7 +629,7 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 				 * Note validation is done on the shard level when DDL propagation
 				 * is enabled. The following eagerly executes some tasks on workers.
 				 */
-				SkipForeignKeyValidationIfConstraintIsFkey(alterTableStmt);
```

Comment:
I'm wondering: could we send `SET citus.skip_constraint_validation TO off` and skip this code path entirely? Not sure, but it feels like we're handling the same issue in a few different ways and that unsurprisingly led to a bug.

off-topic: kind of wondering whether we can get rid of the whole `else` block for the `!EnableDDLPropagation` case

Reply:
So the problem is (if I understand what you mean correctly): We'd need to handle `citus.skip_constraint_validation` differently. Because currently it's only applied during the execution phase, while we need it to also skip the planning phase.

Reply:
Hmz, right, we cannot easily disable planning. Never mind then.